metals/src/main/scala/scala/meta/internal/metals/MetalsLanguageServer.scala

```diff
@@ -1306,20 +1306,24 @@ class MetalsLanguageServer(
         val restartBuildServer = bspSession
           .map { session =>
             if (session.main.isBloop) {
-              bloopServers.ensureDesiredVersion(
-                userConfig.currentBloopVersion,
-                session.version,
-                userConfig.bloopVersion.nonEmpty,
-                old.bloopVersion.isDefined,
-                () => autoConnectToBuildServer
-              )
-
-              bloopServers.ensureDesiredJvmSettings(
-                userConfig.bloopJvmProperties,
-                userConfig.javaHome,
-                () => autoConnectToBuildServer()
-              )
-
+              bloopServers
+                .ensureDesiredVersion(
+                  userConfig.currentBloopVersion,
+                  session.version,
+                  userConfig.bloopVersion.nonEmpty,
+                  old.bloopVersion.isDefined,
+                  () => autoConnectToBuildServer
+                )
+                .flatMap { alreadyRestarted =>
+                  if (!alreadyRestarted)
```

Comment:
[note]
If `ensureDesiredVersion` reconnect to the bloop server, we don't need to `ensureDesiredJvmSettings` because it's already restarted and jvm settings should be already desired state.